huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go

```diff
@@ -409,6 +430,91 @@ func getCBHInstanceList(client *golangsdk.ServiceClient) ([]interface{}, error)
 	return instances, nil
 }
 
+// doPowerAction is a method for CBH instance power doing startup, shutdown and reboot actions.
+func doPowerAction(ctx context.Context, client *golangsdk.ServiceClient, d *schema.ResourceData, action string) error {
+	doActionInstancePath := client.Endpoint + "v2/{project_id}/cbs/instance/{action}"
+	doActionInstancePath = strings.ReplaceAll(doActionInstancePath, "{project_id}", client.ProjectID)
+	jsonBodyMap := map[string]interface{}{
+		"server_id": d.Id(),
+	}
+
+	if action == "SOFT" || action == "HARD" {
+		doActionInstancePath = strings.ReplaceAll(doActionInstancePath, "{action}", "reboot")
+		jsonBodyMap["reboot_type"] = action
+	} else {
+		doActionInstancePath = strings.ReplaceAll(doActionInstancePath, "{action}", action)
+	}
+
+	doActionInstanceOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         jsonBodyMap,
+	}
+	_, err := client.Request("POST", doActionInstancePath, &doActionInstanceOpt)
+	if err != nil {
+		return fmt.Errorf("doing power action (%s) for CBH instance (%s) failed: %s", action, d.Id(), err)
+	}
+
+	if err := waitingForCBHInstanceTaskCompleted(ctx, client, d, d.Timeout(schema.TimeoutCreate)); err != nil {
```

Comment:
The parameter `d.Timeout(schema.TimeoutCreate)` should be input through `doPowerAction`.

Reply:
done.